src/DependencyInjection/Configuration.php

```diff
@@ -154,11 +154,15 @@ public function getConfigTreeBuilder(): TreeBuilder
                                 ->values(['authorization_code', 'implicit'])
                                 ->info('Grant type to use: authorization_code or implicit.')
                                 ->cannotBeEmpty()
-                                ->defaultValue(static function () {
-                                    @trigger_error('The "grant_type" configuration key has been deprecated since 5.3.0 and will be removed in 6.0.', E_USER_DEPRECATED);
+                                ->defaultValue('authorization_code')
+                                ->validate()
+                                    ->ifString()
+                                    ->then(static function ($v) {
+                                        @trigger_error('The "grant_type" configuration key has been deprecated since 5.3.0 and will be removed in 6.0.', E_USER_DEPRECATED);
```

Comment:
```suggestion
                                        @trigger_error('The "grant_type" configuration key has been deprecated since 5.3.0 and will be removed in 6.0. Implicit authentication should not be used in this bundle.', E_USER_DEPRECATED);
```